velox/dwio/parquet/reader/ParquetReader.h

```diff
@@ -78,30 +74,9 @@ class ParquetRowReader : public dwio::common::RowReader {
   // ReaderBase and determines the set of row groups to scan.
   void filterRowGroups();
 
-  // Positions the reader tre at the start of the next row group, as determined
-  // by filterRowGroups().
-  bool advanceToNextRowGroup();
-
-  memory::MemoryPool& pool_;
-  const std::shared_ptr<ReaderBase> readerBase_;
-  const dwio::common::RowReaderOptions options_;
-
-  // All row groups from file metadata.
-  const std::vector<thrift::RowGroup>& rowGroups_;
-
-  // Indices of row groups where stats match filters.
-  std::vector<uint32_t> rowGroupIds_;
-  std::vector<uint64_t> firstRowOfRowGroup_;
-  uint32_t nextRowGroupIdsIdx_;
-  const thrift::RowGroup* FOLLY_NULLABLE currentRowGroupPtr_{nullptr};
-  uint64_t rowsInCurrentRowGroup_;
-  uint64_t currentRowInGroup_;
-
-  std::unique_ptr<dwio::common::SelectiveColumnReader> columnReader_;
-
-  RowTypePtr requestedType_;
-
-  dwio::common::ColumnReaderStatistics columnReaderStats_;
+ protected:
+  class Impl;
+  std::shared_ptr<Impl> impl_;
```

Comment:
Isn't this usually a unique pointer? I suppose there is a reason this isn't a unique pointer?

Reply:
Should be a unique_ptr. Will fix this here https://github.com/facebookincubator/velox/pull/8484